test/e2e/autoscale_test.go

```diff
@@ -245,23 +187,82 @@ func TestAutoscaleUpDownUp(t *testing.T) {
 		},
 		"WaitForAvailablePods")
 	if err != nil {
-		logger.Fatalf(`Waiting for Pod.List to have no non-Evicted pods: %v`, err)
+		ctx.logger.Fatalf("Waiting for Pod.List to have no non-Evicted pods: %v", err)
 	}
 
-	logger.Infof("Scaled down.")
-	logger.Infof("Sending traffic again to verify deployment scales back up")
-	err = generateTraffic(clients, logger, 20, 60*time.Second, domain)
-	if err != nil {
-		t.Fatalf("Error during final scale up: %v", err)
+	ctx.logger.Infof("Scaled down.")
+}
+
+func generateTraffic(ctx *testContext, concurrency int, duration time.Duration) error {
+	var (
+		totalRequests      int
+		successfulRequests int
+		mux                sync.Mutex
+		group              errgroup.Group
+	)
+
+	ctx.logger.Infof("Maintaining %d concurrent requests for %v.", concurrency, duration)
+	for i := 0; i < concurrency; i++ {
+		group.Go(func() error {
+			done := time.After(duration)
+			client, err := pkgTest.NewSpoofingClient(ctx.clients.KubeClient, ctx.logger, ctx.domain, test.ServingFlags.ResolvableDomain)
+			if err != nil {
+				return fmt.Errorf("error creating spoofing client: %v", err)
+			}
+			req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s", ctx.domain), nil)
+			if err != nil {
+				return fmt.Errorf("error creating spoofing client: %v", err)
+			}
+			// ctx.logger.Infof( /* DO NOT SUBMIT */ "sending request to %v", fmt.Sprintf("http://%s", ctx.domain))
+			for {
+				select {
+				case <-done:
+					return nil
+				default:
+					mux.Lock()
+					totalRequests++
+					mux.Unlock()
+					res, err := client.Do(req)
+					if err != nil {
+						ctx.logger.Debugf("error making request %v", err)
+						continue
+					}
+
+					if res.StatusCode != http.StatusOK {
+						ctx.logger.Debugf("non 200 response %v", res.StatusCode)
+						ctx.logger.Debugf("response headers: %v", res.Header)
+						ctx.logger.Debugf("response body: %v", res.Body)
+						continue
+					}
+					mux.Lock()
+					successfulRequests++
+					mux.Unlock()
+				}
+			}
+		})
 	}
-	err = test.WaitForDeploymentState(
-		clients.KubeClient,
-		deploymentName,
-		isDeploymentScaledUp(),
-		"DeploymentScaledUp",
-		2*time.Minute)
-	if err != nil {
-		logger.Fatalf(`Unable to observe the Deployment named %s scaling
-			   up. %s`, deploymentName, err)
+
+	ctx.logger.Infof("Waiting for all requests to complete.")
+	if err := group.Wait(); err != nil {
+		return fmt.Errorf("error making requests for scale up: %v", err)
+	}
+
+	if successfulRequests != totalRequests {
+		return fmt.Errorf("error making requests for scale up. Got %d successful requests. Wanted %d",
+			successfulRequests, totalRequests)
 	}
+
+	ctx.logger.Infof("All %d requests succeeded.", totalRequests)
+	return nil
+}
+
+func TestAutoscaleUpDownUp(t *testing.T) {
```

Comment:
```go
	ctx := setup(t)
	stopChan := DiagnoseMeEvery(15*time.Second, ctx.clients, ctx.logger)
	defer close(stopChan)
	defer tearDown(t, ctx)
 	assertScaleUp(t, ctx)
	assertScaleDown(t, ctx)
	assertScaleUp(t, ctx)
```

Reply:
I put the `defer tearDown` after the `setup` in case the `DiagnoseMeEvery` call panics.